File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        res = 0

        for i in range(len(nums)):
            if nums[i] - 1 not in num_set:
                streak = 1
                curr = nums[i] + 1

                while curr in num_set:
                    streak += 1
                    curr += 1
                
                res = max(res, streak)
            
        return res

File: Data Structures & Algorithms/max-water-container/submission-3.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int res = 0;

        int l = 0;
        int r = heights.size() - 1;

        while (l < r) {
            int min_height = min(heights[r], heights[l]);

            res = max(res, min_height * (r - l));

            if (heights[r] < heights[l]) {
                r -= 1;
            } else {
                l += 1;
            }
        }

        return res;
    }
};
